src/core/core.scale.js

```diff
@@ -410,62 +472,60 @@ module.exports = Element.extend({
 
 		// Don't bother fitting the ticks if we are not showing them
 		if (tickOpts.display && display) {
-			var largestTextWidth = helpers.longestText(me.ctx, tickFont.string, labels, me.longestTextCache);
-			var tallestLabelHeightInLines = helpers.numberOfLabelLines(labels);
-			var lineSpace = tickFont.size * 0.5;
-			var tickPadding = me.options.ticks.padding;
+			var tickFonts = parseTickFontOptions(tickOpts);
+			var labelSizes = computeLabelSizes(me.ctx, tickFonts, ticks, me.longestTextCache);
+			var lineSpace = tickFonts.minor.size * 0.5;
+			var tickPadding = tickOpts.padding;
 
-			// Store max number of lines used in labels for _autoSkip
-			me._maxLabelLines = tallestLabelHeightInLines;
+			// Store the sizes of labels for _autoSkip
+			me._labelSizes = labelSizes;
 
 			if (isHorizontal) {
 				// A horizontal axis is more constrained by the height.
-				me.longestLabelWidth = largestTextWidth;
+				me.longestLabelWidth = labelSizes.widest.width;
 
+				var isRotated = me.labelRotation !== 0;
 				var angleRadians = helpers.toRadians(me.labelRotation);
 				var cosRotation = Math.cos(angleRadians);
 				var sinRotation = Math.sin(angleRadians);
 
-				// TODO - improve this calculation
-				var labelHeight = (sinRotation * largestTextWidth)
-					+ (tickFont.lineHeight * tallestLabelHeightInLines)
+				var labelHeight = sinRotation * labelSizes.widest.width
```

Comment:
`calcRotatedBoxSize` usable here too

Reply:
The dimension of the tick label box used for `labelHeight`, `paddingLeft` and `paddingRight` calculation changes depending on whether it is rotated (need to consider an anchor offset) or not and whether it is at the top or bottom, it's not simple to use `calcRotatedBoxSize` here.